src/platform/i386/exception.c

```diff
@@ -21,12 +21,51 @@ print_regs_state(struct pt_regs *regs)
 	PRINTK("(Exception Error Code-> ORIG_AX: %x)\n", regs->orig_ax);
 }
 
+static void
+fault_regs_save(struct pt_regs *regs, struct thread *thd)
+{
+	copy_all_regs(regs, &(thd->fault_regs));
+	PRINTK("Fault registers saved.\n");
+}
+
 int
-div_by_zero_err_fault_handler(struct pt_regs *regs)
+fault_handler_sinv(struct pt_regs *regs, capid_t cap)
 {
+	struct cos_cpu_local_info *ci       = cos_cpu_local_info();
+	struct thread             *curr_thd = thd_current(ci);
+	struct cap_header         *fh;
+	struct comp_info          *cos_info;
+	thdid_t                    thdid = curr_thd->tid;
+	unsigned long              ip, sp, fault_flag;
+	u32_t                      fault_addr = 0, errcode, eip;
+
 	print_regs_state(regs);
-	die("FAULT: Divide by Zero Error\n\n");
+	fault_regs_save (regs, curr_thd);
+
+	cos_info = thd_invstk_current(curr_thd, &ip, &sp, ci);
+
+	fh = captbl_lkup(cos_info->captbl, cap);
 
+	regs->bx = regs->sp;
+	regs->si = regs->ip;
+	regs->di = fault_addr;
+	regs->dx = cap;
+
+	fault_flag = 1;
+	if (likely(fh->type == CAP_SINV)){
+		sinv_call(curr_thd, (struct cap_sinv *)fh, regs, ci, fault_flag);
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+int
+div_by_zero_err_fault_handler(struct pt_regs *regs)
+{
+	if (!fault_handler_sinv(regs, BOOT_CAPTBL_FLT_DIVZERO)) {
+		die("FAULT: Divide by Zero Error\n");
```

Comment:
can't really have this behavior.  In the short term, this is fine.

We need the next PR to likely be to "hook in" the software fault handlers.  Fault handler not found is one of the faults.